src/qililab/result/stream_results.py

```diff
@@ -0,0 +1,83 @@
+from typing import Any
+import h5py
+import numpy as np
+
+
+class StreamArray:
+    """Allows for real time saving of results from an experiment.
+
+    This class wraps a numpy array and adds a context manager to save results on real time while they are acquired by
+    the instruments.
+
+    Args:
+        shape (tuple): results array shape.
+        path (str): path to save results.
+        loops (dict[str, np.ndarray]): dictionary with each loop name in the experiment as key and numpy array as values.
+    """
+
+    def __init__(self, shape: tuple, path: str, loops: dict[str, np.ndarray]):
+        self.results = np.empty(shape=shape)
+        self.path = path
+        self.loops = loops
+        self.file: h5py.File | None = None
+        self.dataset = None
```

Comment:
Will users use these attributes? If not, let's make them private 🙏

Reply:
I would keep `path` and `loops` public though, because these are the values they give in the init (this is only my humble opinion).

In other words, I expect to have access to the information I provided to a class. Here's an example:

```pycon
>>> a = A(hello="world")
>>> a.hello
world
```

Reply:
agree with this change.